test/spec/shield.js

```diff
@@ -1,64 +1,59 @@
 "use strict";
 
-import chai, { expect } from "chai";
-import * as Shield from "../../src/js/shield.js";
+import { expect } from "chai";
+import {
+  routeParser,
+  shieldPredicate,
+  networkPredicate,
+} from "../../src/js/shield_format";
+import { loadShields } from "../../src/js/shield_defs";
+import {
+  ShieldRenderer,
+  InMemorySpriteRepository,
+} from "@americana/maplibre-shield-generator";
+
+import { HeadlessGraphicsFactory } from "@americana/maplibre-shield-generator/src/headless_graphics";
+
+const shields = loadShields();
+const mockRepo = new InMemorySpriteRepository();
+
+const shieldRenderer = new ShieldRenderer(shields, routeParser)
+  .filterImageID(shieldPredicate)
+  .filterNetwork(networkPredicate)
+  .graphicsFactory(new HeadlessGraphicsFactory())
+  .renderOnRepository(mockRepo);
+
+const handler = shieldRenderer.getStyleImageMissingHandler();
+
+handler({ id: "shield\nBAB=5" });
+handler({ id: "shield\nUS:RI=" });
+handler({ id: "shield\nUS:RI=ABC123" });
+handler({ id: "shield\nUS:RI=Equator" });
+handler({ id: "shield\nrwn=" });
+handler({ id: "foo" });
+
+function isBlankSprite(id) {
+  return mockRepo.getSprite(id).width == 1;
+}
 
 describe("shield", function () {
   describe("#isValidNetwork", function () {
     it("rejects a recreational network", function () {
-      expect(Shield.isValidNetwork("BAB")).to.be.true;
-      expect(Shield.isValidNetwork("rwn")).to.be.false;
+      expect(isBlankSprite("shield\nBAB=5")).to.be.false;
+      expect(isBlankSprite("shield\nrwn=")).to.be.true;
+    });
+    it("rejects other missing image prefixes", function () {
+      expect(mockRepo.hasSprite("foo")).to.be.false;
     });
   });
   describe("#isValidRef", function () {
-    it("rejects a null ref", function () {
-      expect(Shield.isValidRef(null)).to.be.false;
-    });
     it("rejects an empty ref", function () {
-      expect(Shield.isValidRef("")).to.be.false;
+      expect(isBlankSprite("shield\nUS:RI=")).to.be.true;
     });
     it("rejects a long ref", function () {
-      expect(Shield.isValidRef("ABC123")).to.be.true;
-      expect(Shield.isValidRef("Equator")).to.be.false;
-    });
-  });
-
-  describe("#romanizeRef", function () {
```

Comment:
How come these tests are removed?

Reply:
It's in `shieldlib/test/spec/shield.ts`.  Since I moved the functionality into the shield library, I also moved the unit tests into the shield libray.